File: clustering/oop/src/com/graduation/oop/cluster/NewCluster.java
package com.graduation.oop.cluster;

import com.graduation.oop.point.Point;

import java.util.ArrayList;
import java.util.List;

public class NewCluster {
    private Cluster firstCluster;
    private Cluster secondCluster;
    private Double distanceBetween;

    public NewCluster(Cluster firstCluster, Cluster secondCluster, Double distanceBetween) {
        this.firstCluster = firstCluster;
        this.secondCluster = secondCluster;
        this.distanceBetween = distanceBetween;
    }

    public Cluster createCluster(int creationIndex ){
        List<Point<Double>> points = new ArrayList<>(firstCluster.getPoints());
        points.addAll(secondCluster.getPoints());

        ClusterBuilder cb = new ClusterBuilder();
        cb.withCreationIndex(creationIndex)
                .withLeftCluster(this.firstCluster)
                .withRightCluster(this.secondCluster)
                .withPoints(points);

        return cb.build();
    }
}

File: clustering/functional/src/main/scala/data/representation/NewCluster.scala
package data.representation

case class NewCluster[A: Numeric, P[_]](first: Cluster[A, P], second: Cluster[A, P], distanceBetween: A)

object NewCluster {
  def createCluster[A: Numeric, P[_]](newCluster: NewCluster[A, P], creationIndex: Int): Cluster[A, P] =
    Cluster(newCluster.first.points ++ newCluster.second.points,
      creationIndex,
      Some(newCluster.first),
      Some(newCluster.second))
}